design-documents/testing-strategy.md

```diff
@@ -0,0 +1,39 @@
+## Updated Magento Commerce Testing Strategy
+
+### Terms
+
+<!-- Describe any new terms used in the document -->
+
+### Overview
+
+<!-- Describe what we are trying to solve and a few sentence summary of design approach. -->
+
+Despite running 20,000 - 30,000 tests per pull request, we have little to no confidence in the stability of the product. This means that we still need a manual regression process that lasts weeks in the best of cases. 
+
+A sizable portion of this can be attributed to our current high-level testing mentality that prioritizes system-level black box testing over other types of tests. While these types of tests are necessary and good to have, they only provide implicit coverage to the code paths of the concrete implementations that happen to be loaded in the high-level black box tests (they are also expensive/difficult to maintain). We still need explicit coverage for the concrete implementations that extension developers have hard dependencies on. This is true for the simple code paths as well as variations both positive and negative flows.
```

Comment:
What do you mean by high level black box tests, functional tests? In general black box tests are good, and they should be easy to maintain compare to white box tests that knows about implementation. Because every change to implementation (including those that don't change functionality) will break the test. Also, I believe current approach is to use appropriate type of test for each case. It might not be well explained, I can agree with that.